### PR TITLE
Add sample language using xtext 2.14

### PR DESCRIPTION
**Pull request by gregdyke**
ddk.sample.helloworld is the xtext 2.14 equivalent of
ddk.check.test.runtime, and is intended as a sample language for which
DDK can be used

### CODE REVIEW

File: com.avaloq.tools.ddk.sample.helloworld/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,39 @@
+Manifest-Version: 1.0
+Automatic-Module-Name: com.avaloq.tools.ddk.sample.helloworld
+Bundle-ManifestVersion: 2
+Bundle-Name: com.avaloq.tools.ddk.sample.helloworld
+Bundle-Vendor: My Company
+Bundle-Version: 1.0.0.qualifier
+Bundle-SymbolicName: com.avaloq.tools.ddk.sample.helloworld; singleton:=true
+Bundle-ActivationPolicy: lazy
+Require-Bundle: org.eclipse.xtext,
```
Comment (andrewL-avlq) [inline com.avaloq.tools.ddk.sample.helloworld/META-INF/MANIFEST.MF]: also requires org.eclipse.xtext.generator

File: com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -0,0 +1,29 @@
+Manifest-Version: 1.0
+Automatic-Module-Name: com.avaloq.tools.ddk.sample.helloworld.ui
+Bundle-ManifestVersion: 2
+Bundle-Name: com.avaloq.tools.ddk.sample.helloworld.ui
+Bundle-Vendor: My Company
+Bundle-Version: 1.0.0.qualifier
+Bundle-SymbolicName: com.avaloq.tools.ddk.sample.helloworld.ui; singleton:=true
+Bundle-ActivationPolicy: lazy
+Require-Bundle: com.avaloq.tools.ddk.sample.helloworld,
```
Comment (andrewL-avlq) [inline com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF]: Also requires com.avaloq.tools.ddk.xtext

**Comment by andrewL-avlq** [comment]
As long as the issues discussed in the review meeting are addressed, the rest of the workflow migration looks good to me.

File: com.avaloq.tools.ddk.sample.helloworld.ui.test/.classpath [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="src" path="src-gen"/>
+	<classpathentry kind="src" path="xtend-gen"/>
+	<classpathentry kind="src" path="resource"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry kind="output" path="target/classes"/>
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui.test/.classpath]: Should be `<classpathentry kind="output" path="bin"/>`

File: com.avaloq.tools.ddk.sample.helloworld.ui.test/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -0,0 +1,27 @@
+Manifest-Version: 1.0
+Automatic-Module-Name: com.avaloq.tools.ddk.sample.helloworld.tests
+Bundle-ManifestVersion: 2
+Bundle-Name: com.avaloq.tools.ddk.sample.helloworld.tests
+Bundle-Vendor: My Company
+Bundle-Version: 1.0.0.qualifier
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui.test/META-INF/MANIFEST.MF]: Use same version as the rest `2.1.0.qualifier`

File: com.avaloq.tools.ddk.sample.helloworld.ui.test/META-INF/MANIFEST.MF [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+Automatic-Module-Name: com.avaloq.tools.ddk.sample.helloworld.tests
+Bundle-ManifestVersion: 2
+Bundle-Name: com.avaloq.tools.ddk.sample.helloworld.tests
+Bundle-Vendor: My Company
+Bundle-Version: 1.0.0.qualifier
+Bundle-SymbolicName: com.avaloq.tools.ddk.sample.helloworld.ui.test;singleton:=true
+Bundle-ActivationPolicy: lazy
+Require-Bundle: com.avaloq.tools.ddk.sample.helloworld,
+ com.avaloq.tools.ddk.sample.helloworld.ui,
+ org.junit;bundle-version="4.12.0",
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui.test/META-INF/MANIFEST.MF]: No version restriction on junit in the manifest

File: com.avaloq.tools.ddk.sample.helloworld.ui.test/META-INF/MANIFEST.MF [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ org.eclipse.xtext.junit4,
+ org.eclipse.xtext.ui.testing,
+ org.eclipse.xtext.testing,
+ org.eclipse.xtext.xbase.testing,
+ org.eclipse.xtext.xbase.lib;bundle-version="2.14.0",
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui.test/META-INF/MANIFEST.MF]: Remove version restriction from xbase.lib

File: com.avaloq.tools.ddk.sample.helloworld.ui.test/META-INF/MANIFEST.MF [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ org.eclipse.xtext.junit4,
+ org.eclipse.core.runtime,
+ org.eclipse.ui.workbench;resolution:=optional,
+ org.eclipse.xtext.xbase.junit,
+ com.avaloq.tools.ddk.check.runtime.core,
+ com.avaloq.tools.ddk.check.runtime.ui,
+ com.avaloq.tools.ddk.check.core.test,
+ org.eclipse.xtext.ui.testing
+Bundle-RequiredExecutionEnvironment: JavaSE-1.8
+Export-Package: com.avaloq.tools.ddk.sample.helloworld.ui;x-internal=true
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui.test/META-INF/MANIFEST.MF]: Remove `;x-internal=true`

File: com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF [hunk 8]
```diff
@@ -0,0 +1,29 @@
+Manifest-Version: 1.0
+Automatic-Module-Name: com.avaloq.tools.ddk.sample.helloworld.ui
+Bundle-ManifestVersion: 2
+Bundle-Name: com.avaloq.tools.ddk.sample.helloworld.ui
+Bundle-Vendor: My Company
+Bundle-Version: 1.0.0.qualifier
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF]: `2.1.0.qualifier`

File: com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF [hunk 9]
```diff
@@ -0,0 +1,29 @@
+Manifest-Version: 1.0
+Automatic-Module-Name: com.avaloq.tools.ddk.sample.helloworld.ui
+Bundle-ManifestVersion: 2
+Bundle-Name: com.avaloq.tools.ddk.sample.helloworld.ui
+Bundle-Vendor: My Company
+Bundle-Version: 1.0.0.qualifier
+Bundle-SymbolicName: com.avaloq.tools.ddk.sample.helloworld.ui; singleton:=true
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF]: Remove space after `;`

File: com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+Bundle-Name: com.avaloq.tools.ddk.sample.helloworld.ui
+Bundle-Vendor: My Company
+Bundle-Version: 1.0.0.qualifier
+Bundle-SymbolicName: com.avaloq.tools.ddk.sample.helloworld.ui; singleton:=true
+Bundle-ActivationPolicy: lazy
+Require-Bundle: com.avaloq.tools.ddk.sample.helloworld,
+ org.eclipse.xtext.ui,
+ org.eclipse.xtext.ui.shared,
+ org.eclipse.xtext.ui.codetemplates.ui,
+ org.eclipse.ui.editors;bundle-version="3.5.0",
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF]: Remove `;bundle-version="3.5.0"`

File: com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+ org.eclipse.ui.ide;bundle-version="3.5.0",
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF]: Remove `;bundle-version="3.5.0"`

File: com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+ org.eclipse.ui,
+ org.eclipse.compare,
+ org.eclipse.xtext.builder,
+ org.eclipse.xtext.xbase.lib;bundle-version="2.14.0",
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF]: Remove `;bundle-version="2.14.0"`

File: com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+ org.eclipse.xtend.lib;resolution:=optional,
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui/META-INF/MANIFEST.MF]: Why is dependency to `org.eclipse.xtend.lib` optional?

File: com.avaloq.tools.ddk.sample.helloworld.ui/build.properties [hunk 14]
```diff
@@ -0,0 +1,7 @@
+source.. = src/,\
+           src-gen/,\
+           xtend-gen/
+bin.includes = .,\
+               META-INF/,\
+               plugin.xml
+bin.excludes = **/*.xtend
```
Comment (rmitin) [inline com.avaloq.tools.ddk.sample.helloworld.ui/build.properties]: Remove the exclude

**Pull request merged**